driver/src/test/java/org/neo4j/driver/internal/connector/socket/ChunkedInputTest.java

```diff
@@ -150,4 +152,30 @@ public void canReadAllIntegerSizes() throws Exception
         assertEquals( 0,                 in.readInt() );
         assertEquals( Integer.MIN_VALUE, in.readInt() );
     }
-}
\ No newline at end of file
+
+    @Test
+    public void shouldNotReadMessageEndingWhenByteLeftInBuffer()
```

Comment:
This test is awesome, no need to change it, but noting that Nigel showed me an even awesomer way to write these kinds of tests, using the `ExpectedException` JUnit rule, you can do something like;

    @Rule public ExpectedException exception = ExpectedException.none();

    @Test public void blah() throws Exception {
        // given
        some stuff

        // expect
        exception.expect( "It didn't work oh no");

        // when
        ch.messageBoundaryHook().run();
    }